src/Entity/Engine.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace Drupal\elastic_appsearch\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\elastic_appsearch\ElasticSearchInterface;
+use Drupal\elastic_appsearch\Utility\Database;
+
+/**
+ * Defines the Engine entity.
+ *
+ * @ConfigEntityType(
+ *   id = "engine",
+ *   label = @Translation("Engine"),
+ *   handlers = {
+ *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
+ *     "list_builder" = "Drupal\elastic_appsearch\EngineListBuilder",
+ *     "form" = {
+ *       "add" = "Drupal\elastic_appsearch\Form\EngineForm",
+ *       "edit" = "Drupal\elastic_appsearch\Form\EngineForm",
+ *       "delete" = "Drupal\elastic_appsearch\Form\EngineDeleteForm",
+ *       "schema" = "Drupal\elastic_appsearch\Form\FieldSchemaForm",
+ *     },
+ *     "route_provider" = {
+ *       "html" = "Drupal\elastic_appsearch\EngineHtmlRouteProvider",
+ *     },
+ *   },
+ *   config_prefix = "engine",
```

Comment:
Also, the config prefix should start with the module name. I think it should be `elastic_appsearch.engine`. What do you think?